src/shared/sol-util-file.c

```diff
@@ -86,6 +86,10 @@ sol_util_write_file(const char *path, const char *fmt, ...)
     va_list ap;
     int ret;
 
+    ret = access(path, 0);
+    if (ret < 0)
+        return ret;
+
```

Comment:
Hi,

Unfortunately, this log doesn't contain line numbers =/ To get them, run valgrind on a debug build or build Soletta with `make CFLAGS='-g'`.
Nevertheless, by putting the pieces together, I guess that the real issue is that iio device wasn't created at all - I remember a warning that you reported elsewhere about it. So I believe that the real fix is to see where iio device creation failed (some call to `sol_iio_open`), and abort the flow there.
I also understood that device creation is failing because it's attempting to do a buffered reading on a device that does not support buffers. We could fallback to 'not buffered' readings as well, instead of aborting the flow.
Either way, `sol_util_write_file` is not wrong, there's no need to change it, regarding this issue. If it can not create the file, it will just return an error. Look at this line on valgrind log:

     ==2032== at 0x4ECFED5: sol_iio_read_channel_value (in /usr/lib/libsoletta.so)

The error is on `sol_iio_read_channel_value`, not on `sol_util_write_file`. You can check the line if you build Soletta with debug information before running on valgrind.